src/reformat_libyuv.c

```diff
@@ -461,6 +501,275 @@ avifResult avifImageYUVToRGBLibYUV(const avifImage * image, avifRGBImage * rgb)
     return AVIF_RESULT_NOT_IMPLEMENTED;
 }
 
+avifResult avifImageYUVToRGBLibYUV10bpc(const avifImage * image,
+                                        avifRGBImage * rgb,
+                                        const struct YuvConstants * matrixYUV,
+                                        const struct YuvConstants * matrixYVU)
+{
+    // See if the current settings can be accomplished with libyuv, and use it (if possible).
+
+    assert((image->depth == 10) && (rgb->depth == 8));
+
+    // This following section might be a bit complicated to audit without a bit of explanation:
+    //
+    // libavif uses byte-order when describing pixel formats, such that the R in RGBA is the lowest address,
+    // similar to PNG. libyuv orders in word-order, so libavif's RGBA would be referred to in libyuv as ABGR.
+    // In addition, swapping U and V in any of these calls, along with using the Yvu matrix instead of Yuv matrix,
+    // swaps B and R in these orderings as well. This table summarizes this block's intent:
+    //
+    // libavif format        libyuv Func     UV matrix (and UV argument ordering)
+    // --------------------  -------------   ------------------------------------
+    // AVIF_RGB_FORMAT_RGB   n/a             n/a
+    // AVIF_RGB_FORMAT_BGR   n/a             n/a
+    // AVIF_RGB_FORMAT_BGRA  *ToARGBMatrix   matrixYUV
+    // AVIF_RGB_FORMAT_RGBA  *ToARGBMatrix   matrixYVU
+    // AVIF_RGB_FORMAT_ABGR  *ToRGBAMatrix   matrixYUV
+    // AVIF_RGB_FORMAT_ARGB  *ToRGBAMatrix   matrixYVU
+
+    if (rgb->format == AVIF_RGB_FORMAT_BGRA) {
+        // AVIF_RGB_FORMAT_BGRA  *ToARGBMatrix   matrixYUV
+
+#if LIBYUV_VERSION >= 1780
+        if (image->yuvFormat == AVIF_PIXEL_FORMAT_YUV444) {
+            if (I410ToARGBMatrix((const uint16_t *)image->yuvPlanes[AVIF_CHAN_Y],
+                                 image->yuvRowBytes[AVIF_CHAN_Y] / 2,
+                                 (const uint16_t *)image->yuvPlanes[AVIF_CHAN_U],
+                                 image->yuvRowBytes[AVIF_CHAN_U] / 2,
+                                 (const uint16_t *)image->yuvPlanes[AVIF_CHAN_V],
+                                 image->yuvRowBytes[AVIF_CHAN_V] / 2,
```

Comment:
@tongyuantongyu Please confirm that we must divide `image->yuvPlanes[AVIF_CHAN_Y]` by 2 to get the `src_stride_y` value for libyuv.

Reply:
Yes I confirm it. Libyuv treats stride as ptrdiff_t and add it to uint16_t* pointer directly.